internal/common/proto/protoutil_test.go

```diff
@@ -70,6 +70,35 @@ func TestMustMarshallAndCompress(t *testing.T) {
 	assert.Equal(t, compressedMsg, bytes)
 }
 
+func TestHashMany(t *testing.T) {
+	input1 := &schedulerobjects.JobSchedulingInfo{Lifetime: 1}
+	input2 := &schedulerobjects.JobSchedulingInfo{Lifetime: 2}
+	input3 := &schedulerobjects.JobSchedulingInfo{Lifetime: 3}
+	hash1, err := HashMany([]*schedulerobjects.JobSchedulingInfo{input1, input2})
+	require.NoError(t, err)
+	require.Equal(t, 20, len(hash1))
+	hash2, err := HashMany([]*schedulerobjects.JobSchedulingInfo{input1, input2})
+	require.NoError(t, err)
+	require.Equal(t, 20, len(hash2))
+	hash3, err := HashMany([]*schedulerobjects.JobSchedulingInfo{input1, input3})
+	require.NoError(t, err)
+	require.Equal(t, 20, len(hash3))
+	assert.Equal(t, hash1, hash2)
+	assert.NotEqual(t, hash1, hash3)
+}
+
+func TestHash_NoCollisions(t *testing.T) {
+	hashes := make(map[string]bool)
+	for i := 0; i < 1000000; i++ {
```

Comment:
Do we really want to run one million assertions for this on every test invocation?

Reply:
it's a fair point- although it takes about 2 seconds on my laptop so shouldn't be too bad.